### PR TITLE
Python Build Dependencies Survey 20231024

### PR DESCRIPTION
**Pull request by MingcongBai**
Topic Description
-----------------

This topic surveys for Python build dependency issues (such as implicitly specified `setuptools` build dependencies). It also addresses some build failures found with Python 3.10.

This topic is part of the effort to merge #4701.

Package(s) Affected
-------------------

See "Files changed."

Security Update?
----------------

No

Build Order
-----------

```
capstone meson sphinx-rtd-theme net-snmp beaker cachecontrol chardet charset-normalizer crashtest enum34 cryptography tomli flit html5lib-python secretstorage keyring m2crypto markupsafe mako py-filelock sip pyqt5 pysocks pytest pyudev schema semantic-version toolbelt toposort apparmor platformdirs python-distutils-extra catfish mutagen urwid s-tui setconf bcrypt cairocffi click isodate paramiko kiwisolver mistune pefile pysmbc python-systemd python-xapp rdflib setuptools-python3 tinycss2
```

Test Build(s) Done
------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

### CODE REVIEW

File: lang-python/rdflib/spec [hunk 1]
```diff
@@ -1,5 +1,5 @@
-VER=4.2.2
-REL=3
+VER=6.2.0
+REL=1
```
Comment (liushuyu) [inline lang-python/rdflib/spec]:
```suggestion
```

File: lang-python/tinycss2/autobuild/defines [hunk 2]
```diff
@@ -1,6 +1,7 @@
 PKGNAME=tinycss2
 PKGSEC=python
 PKGDEP="webencodings"
+BUILDDEP="pip setuptools"
```
Comment (liushuyu) [inline lang-python/tinycss2/autobuild/defines]: Questionable build dependency: `pip`.

File: lang-python/pyqt5/spec [hunk 3]
```diff
@@ -1,5 +1,5 @@
 VER=5.15.5
-REL=2
+REL=4
```
Comment (liushuyu) [inline lang-python/pyqt5/spec]:
```suggestion
REL=3
```

File: lang-python/docutils/spec [hunk 4]
```diff
@@ -1,4 +1,4 @@
-VER=0.19
+VER=0.17.1
```
Comment (liushuyu) [inline lang-python/docutils/spec]: Why this downgrade is required?

**Review by liushuyu — COMMENTED** [review]
Mostly LGTM. Some comments need to be addressed.

**Review by eatradish — APPROVED** [review]

**Pull request merged**